src/main/java/com/appland/appmap/AppmapPluginExtension.java

```diff
@@ -0,0 +1,110 @@
+package com.appland.appmap;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.process.JavaForkOptions;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class holds the configuration parameters read from the target project gradle.build
+ */
+public class AppmapPluginExtension {
+
+    public static final String DEFAULT_OUTPUT_DIRECTORY = "build/appmap";
+    protected final Project project;
+    private final Logger LOGGER = Logger.getLogger("com.appland.appmap");
+    private final Configuration agentConf;
+    private final RegularFileProperty configFile;
+    private final DirectoryProperty outputDirectory;
+    private boolean skip = false;
+    private String debug = "info"; //Enable debug flags as a comma separated list. Accepts: info, hooks, http, locals Default: info
+    private String debugFile = "target/appmap/agent.log";
+    private int eventValueSize = 1024; //Specifies the length of a value string before truncation. If 0, truncation is disabled.
+
+    public AppmapPluginExtension(Project project, Configuration agentConf, JavaForkOptions task) {
+        this.project = project;
+        this.agentConf = agentConf;
+        this.configFile = project.getObjects().fileProperty().fileValue(new File("appmap.yml"));
+        this.outputDirectory = project.getObjects().directoryProperty().fileValue(new File(DEFAULT_OUTPUT_DIRECTORY));
+        try {
+            LOGGER.setLevel(Level.parse(debug.toUpperCase()));
```

Comment:
`appmap.debug` won't match up with a logger level (other than `INFO`). It's expected to be a comma separated list of subsystems to enable verbose logging (`hooks,http,locals`).

Reply:
You are right, but then I should just put logger at INFO level, and omit this until an issue identifying what to log at hooks, HTTP and locals should log, right?